<!DOCTYPE html>
<html lang="en" dir="ltr">

      <head>
        <meta charset="utf-8">
        <title>FusionESP</title>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family==Rancho&effect=shadow-multiple' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link href='https://fonts.googleapis.com/css?family=Tangerine' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="{{ url_for('static', filename='css/style.css') }}">

		<style>
			.content {
			  max-width: 1000px;
			  margin: auto;
			}

			</style>
    </head>

<body>
    <div class="content">
        <h1>FusionESP: Improved enzyme-substrate pair prediction by fusing protein and chemical knowledge</h1>
        <p style="font-size:70%">The webserver is the implementation of the paper "" The original codes for the model development are available at https://github.com/dzjxzyd/FusionESP</p>
	<p style="font-size:80%">Notice: a confidence score will be given for you, which reflect how confidence the model think its output. The model deployed in this web server is the  CLR_ESP-XL with ESM-2-1280 and MolFormer trained on both phylogenetic evidence-based and experimental evidence-based datasets (ACC = 94.56%	AUC = 0.9635	MCC = 0.8572)</p>
        <form action="{{url_for('predict')}}" method="post">
            <input type="text" name="Enzyme_sequence" placeholder="enzyme seuqence" required="required" />
            <input type="text" name="molecule_SMILES" placeholder="SMILES format" required="required" />
            <button type="submit" class="btn btn-primary btn-block btn-large">Run</button>
        </form>
        <p style="color:cyan"> {{ prediction_text }} </p>

        <br/>
        <br/>
        <p style="font-size:80%">Large-scale output version: 1. Prepare your files (xls, xlsx) and click “Choose File” for uploading → 2. Choose a model for classification  → 3. Download the results. </p>
        <form action="{{url_for('pred_with_file')}}" method="post" enctype = "multipart/form-data">
        <br>
            <label for="Peptide_sequences">Choose a file to upload</label>
            <input type="file" name="Peptide_sequences" id="Peptide_sequences" accept=".xls,.xlsx"/>
            <button type="submit" class="btn btn-primary btn-block btn-large">Run</button>
        </form>
        <h4>Usage of the webserver:</h4>
        <p style="font-size:80%"> Example for “Quick output version” : </p>
        <p style="font-size:70%">1. Insert a enzyme protein sequence (in single amino acid code), and a SMILES representation of molecules (canonical SMILES) →   →  →  2. Click “Run”→   →  → 4. The result will be returned in seconds below the “Run” button</p>
        <p style="font-size:70%">Notice: it also support multiple sequence at the same time. Just input as “VPP,IPP,CCL,AGR” (sequences are separated by comma, no space)</p>
        <p style="font-size:80%"> Example for “Large-scale output version:” : </p>
        <p style="font-size:70%">1. Prepare your xls, xlsx, txt or fasta files  →  →  → 2. Upload the file through “Choose File” botton  →   →  → 3. Click “Run” →   →  → 4. It will automatically download your results. </p>
        <p style="font-size:70%">Notice: File preparation should follow the examples under this repository https://github.com/dzjxzyd/FusionESP_server_1280/tree/main/input </p>

        <img src="static/css/Whole architecture.jpg" alt="Whole architecture" width="640" height="380" style="float:left">
    </div>
</body>
</html>
